src/player.cpp

```diff
@@ -644,8 +717,13 @@ int Player::countAudioSource(unsigned int soundHash)
         return soloud.countAudioSource(*as);
     }
 
-    // if (s->soundType == TYPE_WAVSTREAM)
-    SoLoud::AudioSource *as = static_cast<SoLoud::WavStream *>(s->sound.get());
+    if (s->soundType == TYPE_WAVSTREAM)
+    {
+        SoLoud::AudioSource *as = static_cast<SoLoud::WavStream *>(s->sound.get());
+        return soloud.countAudioSource(*as);
+    }
+
+    SoLoud::AudioSource *as = static_cast<SoLoud::BufferStream *>(s->sound.get());
```

Comment:
[nit] I think it makes sense now to `switch` over `s->soundType` instead of having if statements and the last possibility outside an if statement.

Reply:
Done.